### ISSUE TITLE
Unable to run hack/update-codegen.sh

### ISSUE THREAD

**Issue by kannon92** [kubernetes/kubernetes#116094]
### What happened?

I am working on a PR where I am removing some legacy code from a API.  

I believe I have to run some scripts to generate code.

When I run `make update`

I get the following error:

```
Running update-codegen
go version go1.20.1 linux/amd64
/home/ec2-user/Work/GIT/kubernetes-kevin/hack/update-codegen.sh: line 1008: flags_to_pass[@]: unbound variable
```

### What did you expect to happen?

I expect the script to work and run without error.

### How can we reproduce it (as minimally and precisely as possible)?

Run hack/update-codegen.sh or run `make update`

### Anything else we need to know?

I am on the latest build.

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by liggitt** [comment]
cc @thockin 

are you synced to HEAD of master? what version of bash are you running?

**Comment by kannon92** [comment]
> cc @thockin
> 
> are you synced to HEAD of master? what version of bash are you running?

I am synced to HEAD of master and my bash version is  version 4.2.46(2).

**Comment by tzneal** [comment]
Pretty sure @liggitt identified it correctly as a bash version issue.  I ran into this issue and it was due to using an older bash.  I know that 4.2.46 (which is used in AL2) is too old, but I'm not sure exactly what the minimum version is.

If you are using AL2 for development, this will drop in a compatible version:

```
sudo yum install -y gcc util-linux-user
cd /tmp
git clone https://github.com/bminor/bash.git
cd bash
git checkout bash-5.2
./configure
make -j8
sudo rm /usr/bin/bash
sudo cp bash /usr/bin/bash
```

**Comment by thockin** [comment]
Let me see if there's a way to make old-bash shut up.  Sadly I do not have an old version to try against - @kannon92 - are you willing to type commands into your shell for me?

**Comment by kannon92** [comment]
> Let me see if there's a way to make old-bash shut up. Sadly I do not have an old version to try against - @kannon92 - are you willing to type commands into your shell for me?

Yep, you just caught me before I purged my old environment with @tzneal suggestions.

**Comment by kannon92** [comment]
@thockin would a slack conversation be easier?

**Comment by thockin** [comment]
maybe but I am juggling meetings :)

**Comment by tzneal** [comment]
There are a few problems I ran into before just deciding to upgrade to bash v5, but I think this initial thing was a bug fixed in bash 4.4+:

From, https://lists.gnu.org/archive/html/bug-bash/2016-08/msg00123.html

> a.  Using address@hidden or ${a[*]} with an array without any assigned elements 
when the nounset option is enabled no longer throws an unbound variable error.

So on 4.2, this fails:

```
#!/usr/bin/env bash

set -o nounset

flags_to_pass=()
echo "${flags_to_pass[@]}"
```

**Comment by thockin** [comment]
working on a patch
